themes/dotmatrix/dotmatrix.theme

```diff
@@ -0,0 +1,1053 @@
+
+    DOTMATRIX_COLOR_PATH_LAST=( ${DOTMATRIX_COLOR_PATH_LAST[@]+"${DOTMATRIX_COLOR_PATH_LAST[@]}"} )
+    [[ ${#DOTMATRIX_COLOR_PATH_LAST[@]} == 0 ]] && DOTMATRIX_COLOR_PATH_LAST=(15 239 1 0   0  15)
+
+
+    DOTMATRIX_COLOR_LINE=( ${DOTMATRIX_COLOR_LINE[@]+"${DOTMATRIX_COLOR_LINE[@]}"} )
+    [[ ${#DOTMATRIX_COLOR_LINE[@]} == 0 ]] && DOTMATRIX_COLOR_LINE=(244 -1 0 0 15 -1)
+
+    # DOTMATRIX_COLOR_LINE_START=( ${DOTMATRIX_COLOR_LINE_START[@]+"${DOTMATRIX_COLOR_LINE_START[@]}"} )
+    # [[ ${#DOTMATRIX_COLOR_LINE_START[@]} == 0 ]] && DOTMATRIX_COLOR_LINE_START=(255 -1 1 0 15 -1)
+
+    DOTMATRIX_COLOR_LINE_END=( ${DOTMATRIX_COLOR_LINE_END[@]+"${DOTMATRIX_COLOR_LINE_END[@]}"} )
+    [[ ${#DOTMATRIX_COLOR_LINE_END[@]} == 0 ]] && DOTMATRIX_COLOR_LINE_END=(232 -1 1 0 15 -1)
+
+
+    DOTMATRIX_COLOR_MARK=( ${DOTMATRIX_COLOR_MARK[@]+"${DOTMATRIX_COLOR_MARK[@]}"} )
+    [[ ${#DOTMATRIX_COLOR_MARK[@]} == 0 ]] && DOTMATRIX_COLOR_MARK=(15 -1 0 0 15 -1)
+
+    DOTMATRIX_COLOR_SEP_DARK=( ${DOTMATRIX_COLOR_SEP_DARK[@]+"${DOTMATRIX_COLOR_SEP_DARK[@]}"} )
+    [[ ${#DOTMATRIX_COLOR_SEP_DARK[@]} == 0 ]] && DOTMATRIX_COLOR_SEP_DARK=(  0 -2 0 0  0 -2)
+
+    DOTMATRIX_COLOR_SEP_FAIR=( ${DOTMATRIX_COLOR_SEP_FAIR[@]+"${DOTMATRIX_COLOR_SEP_FAIR[@]}"} )
+    [[ ${#DOTMATRIX_COLOR_SEP_FAIR[@]} == 0 ]] && DOTMATRIX_COLOR_SEP_FAIR=(244 -2 0 0  0 -2)
+
+    DOTMATRIX_COLOR_SEP_LITE=( ${DOTMATRIX_COLOR_SEP_LITE[@]+"${DOTMATRIX_COLOR_SEP_LITE[@]}"} )
+    [[ ${#DOTMATRIX_COLOR_SEP_LITE[@]} == 0 ]] && DOTMATRIX_COLOR_SEP_LITE=( 15 -2 0 0 15 -2)
+
+
+    __dotmatrix_activate_display
+    __dotmatrix_activate_connection
+    __dotmatrix_activate_user
+    __dotmatrix_activate_multiplexer
+    __dotmatrix_activate_hostname
+}
+
+
+_lp_dotmatrix_theme_directory() {
+    lp_terminal_format ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+    __dotmatrix_directory_FS
+    __dotmatrix_directory_path
+}
+
+
+_lp_dotmatrix_theme_prompt() {
+
+    local header inline prompt
+
+    # Arguments: fg, bg, bold, underline, fallback_fg, fallback_bg
+    # Color codes can be any integer in 256 XTerm's system colors in [0–255].
+    # Special color codes:
+    #   -1 : no color
+    #   -2 : previously set color
+    #   -3 : swap fg and bg
+    local lp_terminal_format
+
+    ####################################################################
+    # HEADER (LEFT)
+    ####################################################################
+
+    header=""
+    header+="${_DOTMATRIX_DISPLAY}"
+    header+="${_DOTMATRIX_CONNECTION}"
+    header+="${_DOTMATRIX_USER}"
+    header+="${_DOTMATRIX_MULTIPLEXER}"
+    header+="${_DOTMATRIX_HOSTNAME}"
+    header+="${_DOTMATRIX_FS}"
+    header+="${_DOTMATRIX_PATH} "
+
+    if _lp_dirstack; then
+        __dotmatrix_make_sep_plug ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+        header+="${dotmatrix_sep}"
+        __dotmatrix_make_field $lp_dirstack ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
```

Comment:
`$lp_dirstack` needs to be quoted here.

See https://github.com/nojhan/liquidprompt/wiki/Shell-standards#quoting
> There are almost no scenarios where quoting a variable is incorrect, so unless you know for sure that not quoting it is safe, just quote it!

Reply:
Very sorry… I did a round trip on the code and tried to be consistent.